edb/server/mng_port/edgecon.pyx

```diff
@@ -1120,7 +1177,14 @@ cdef class EdgeConnection:
         compiled_query = await self._parse(eql, query_req)
 
         buf = WriteBuffer.new_message(b'1')  # ParseComplete
-        buf.write_int16(0)  # no headers
+
+        buf.write_int16(1)
+        buf.write_int16(SERVER_HEADER_CAPABILITIES)
+        buf.write_int32(sizeof(uint64_t))
+        buf.write_int64(<int64_t>(
+            PUBLIC_CAPABILITIES & compiled_query.query_unit.capabilities
+        ))
```

Comment:
@elprans I'm gonna merge this now, but we need to have a quick discussion about common header serialization format. Did you want to use type descriptors for that?

Reply:
>  Did you want to use type descriptors for that?

I think that was one of the options, but I don't recall if there was a more detailed plan of action.  For example, would we transmit the type descriptor for each header, or would that be hardcoded?

In any case, I think the current string-based form with arbitrary rules and ad-hoc validation is quite bad.